gnovm/pkg/gnolang/joeson_test.go

```diff
@@ -0,0 +1,776 @@
+		}
+	} else {
+		return errors.New("expecting BasicLitExpr")
+	}
+}
+
+func (v parsesAsNil) satisfies(ast j.Ast, expectation expectation) error {
+	if ast != nil {
+		return errors.New(fmt.Sprintf("was expecting nil, got %q", ast.String()))
+	}
+	return nil
+}
+
+func (v isBasicLit) satisfies(ast j.Ast, expectation expectation) error {
+	if expr, ok := ast.(*BasicLitExpr); ok {
+		if expr.Kind != v.kind {
+			return errors.New(fmt.Sprintf(
+				"was expecting Kind=%s for &BasicLitExpr, got %s",
+				v.kind,
+				expr.Kind,
+			))
+		}
+	} else {
+		return errors.New(fmt.Sprintf(
+			"was expecting &BasicLitExpr (%v), got %s",
+			v.kind,
+			reflect.TypeOf(ast).String(),
+		))
+	}
+	return nil
+}
+
+func (v isSelectorExpr) satisfies(ast j.Ast, expectation expectation) error {
+	if _, ok := ast.(*SelectorExpr); !ok {
+		return errors.New(fmt.Sprintf(
+			"was expecting &SelectorExpr, got %s",
+			reflect.TypeOf(ast).String(),
+		))
+	}
+	return nil
+}
+
+func (v isNameExpr) satisfies(ast j.Ast, expectation expectation) error {
+	if _, ok := ast.(*NameExpr); !ok {
+		return errors.New(fmt.Sprintf(
+			"was expecting &NameExpr, got %s",
+			reflect.TypeOf(ast).String(),
+		))
+	}
+	return nil
+}
+
+func (v isCallExpr) satisfies(ast j.Ast, expectation expectation) error {
+	if _, ok := ast.(*CallExpr); !ok {
+		return errors.New(fmt.Sprintf(
+			"was expecting &CallExpr, got %s",
+			reflect.TypeOf(ast).String(),
+		))
+	}
+	return nil
+}
+
+func (v errorIs) satisfies(ast j.Ast, expectation expectation) error {
+	if !j.IsParseError(ast) {
+		return errors.New(fmt.Sprintf(
+			"was expecting error %q, got result %q", v.string, ast.String()))
+	}
+	if v.string != "" && strings.TrimPrefix(ast.String(), "ERROR ") != v.string {
+		return errors.New(fmt.Sprintf(
+			"although we got a parse error as expected, were expecting %q"+
+				", got %q", v.string, ast.String()))
+	}
+	return nil
+}
+
+func (v errorContains) satisfies(ast j.Ast, expectation expectation) error {
+	if !j.IsParseError(ast) {
+		return errors.New(fmt.Sprintf(
+			"was expecting error %q, got %q", v.string, ast.String()))
+	}
+	if !strings.Contains(ast.String(), v.string) {
+		return errors.New(fmt.Sprintf(
+			"parse error as expected, but expecting error to contain \"%s\", "+
+				"got \"%s\" instead", v.string, ast.String()))
+	}
+	return nil
+}
+
+func (noError) satisfies(ast j.Ast, expectation expectation) error {
+	if j.IsParseError(ast) {
+		return errors.New(fmt.Sprintf(
+			"unexpected ParseError, was expecting %s", expectation.brief()))
+	}
+	return nil
+}
+
+func (t isType) satisfies(ast j.Ast, expectation expectation) error {
+	theType := fmt.Sprintf("%T", ast)
+	if !strings.HasSuffix(theType, t.string) {
+		return errors.New(fmt.Sprintf("type should have been %s, not %s",
+			t.string, theType))
+	}
+	return nil
+}
+
+// [deprecated] Pseudo evaluation of expressions to test parsing
+// is now deprecated in favor of usage of bxPolishNotationIs{}.
```

Comment:
```suggestion
// is now deprecated in favor of usage of bxPolishNotationIs{}.
```
// is now deprecated in favor of bxPolishNotationIs{}.